### ISSUE TITLE
Issue with mail server

### ISSUE THREAD

**Issue by glemaitre** [paris-saclay-cds/ramp-board#526]
The server can hang when sending an email to the participant.
With @agramfort, we think that it could come from the fact that to send an email, we connect to the mail server and send the email. However, this procedure is blocking the rest of the execution. We should migrate to asynchronous processing.

Some details with Flask integration are given there: https://stackoverflow.com/questions/40326651/flask-mail-sending-email-asynchronously-based-on-flask-cookiecutter

**Comment by rth** [comment]
I agree that sending emails is likely a bottleneck. The question is whether it's slow by default, or if we are also getting rate limited by GMail. For instance https://stackoverflow.com/a/19784201/1791279 indicates that there should be some delays between sent mails. 

https://support.google.com/mail/answer/81126 recommends,

> If you send many messages, we recommend you:
> 
>    Send mail at a consistent rate. Avoid sending mail in bursts.
>    Start with a low sending volume, then slowly increase the volume over time.

And in that case if there are 30 users that try to sign up at the same time, implementing async requests with increase the rate of requests and make more likely that emails get blocked by the SMTP server. 

Another solution could be to attach a Queue and a long running thread to the flask app. Then when sending email, send the email to the queue and let the thread look at the queue and send it, which avoid sending in burst. It does mean though that users may receive the email after a few minutes. If there are multiple emails in the queue, it's also possible to re-use SMTP connections (https://pythonhosted.org/Flask-Mail/#bulk-emails) which might be faster, but I'm not sure about rate limits. It might also be a bit easier to test, as we keep the thread object. 

The other solution could be to change the SMTP provider, but I imagine all implement some rate limit to some extent.

**Comment by agramfort** [comment]
it turns out that we already use  hover.com as smtp server. It's the
service used for the domain name.

>
